internal/resource/url.go

```diff
@@ -287,6 +293,28 @@ func (f *Fetcher) fetchFromHTTP(u url.URL, dest io.Writer, opts FetchOptions) er
 		}
 	}
 
+	if opts.LocalPort != nil {
+		var (
+			d net.Dialer
+			p int
+		)
+
+		// Assert that the port is not already used.
+		for {
+			p = opts.LocalPort()
+			l, err := net.Listen("tcp4", fmt.Sprintf(":%d", p))
+			if err != nil && errors.Is(err, syscall.EADDRINUSE) {
+				continue
```

Comment:
Minor: won't this busy loop here? Should we add a delay between tries?

Though maybe a cleaner approach is to lower the concept of "use privileged client port" down into the Fetcher directly so that it can e.g. try other ports in the list.

But anyway, in practice since the only user of this is the Scaleway code and it's requesting a privileged port, it's likely the only way we hit this is if there's e.g. a daemon listening in the initrd, which there aren't.

Not worth blocking on!